src/UseCases/AddComment/AddCommentUseCase.php

```diff
@@ -63,9 +72,18 @@ private function requestIsAllowed( AddCommentRequest $addCommentRequest ): bool
 	private function newCommentFromRequest( AddCommentRequest $request ): DonationComment {
 		return new DonationComment(
 			$request->getCommentText(),
-			$request->isPublic(),
+			$this->commentCanBePublic( $request ),
 			$request->getAuthorDisplayName()
 		);
 	}
 
+	private function commentCanBePublic( AddCommentRequest $request ): bool {
+		return $request->isPublic()
+			&& $this->commentTextPassesValidation( $request->getCommentText() );
```

Comment:
Again, this removes the original intent of users whether or the comment should be public. What about we add another filter for moderated (and deleted/cancelled) donations in the `CommentRepository` to prevent that stuff from ending up in the public list of comments?